### PR TITLE
fix: Validate and sanitize certificate for logpipeline

### PR DESCRIPTION
**Pull request by rakesh-garimella**
## Description

Changes proposed in this pull request (what was done and why):

- Sometimes there are extra escape characters in the tls secret and key. In such case identify it and fix it by removing extra escape characters.
- Validate certificate for trace/metric pipelines as well
Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/670

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: internal/secretref/secret_ref.go [hunk 1]
```diff
@@ -28,6 +30,19 @@ func ReferencesNonExistentSecret(ctx context.Context, client client.Reader, gett
 	return false
 }
 
+func ValidateAndSanitizeTLSSecret(tlsCert, tlsKey []byte) ([]byte, []byte) {
+	_, err := tls.X509KeyPair(tlsCert, tlsKey)
+	if err != nil {
+		if noPemDataFoundError(err) {
```
Comment (chrkl) [inline internal/secretref/secret_ref.go:36]: I would do the replacement in the case of any error. Relying on the message is not very robust and this replacement should usually  not cause any harm.

File: internal/secretref/secret_ref.go [hunk 2]
```diff
@@ -28,6 +30,19 @@ func ReferencesNonExistentSecret(ctx context.Context, client client.Reader, gett
 	return false
 }
 
+func ValidateAndSanitizeTLSSecret(tlsCert, tlsKey []byte) ([]byte, []byte) {
```
Comment (chrkl) [inline internal/secretref/secret_ref.go:33]: Not related to secret references. Let's move this to the package where it's used.

File: internal/secretref/secret_ref.go [hunk 3]
```diff
@@ -28,6 +30,29 @@ func ReferencesNonExistentSecret(ctx context.Context, client client.Reader, gett
 	return false
 }
 
+func SanitizeTlSValueOrSecret(secretData map[string][]byte, tlsKeyVariable, tlsCertVariable string) map[string][]byte {
```
Comment (chrkl) [inline internal/secretref/secret_ref.go:33]: I think we can improve multiple things about this function:
1. Map arguments are references in Go, so no need to return the same map
2. Abou the naming
  a. I cannot distinguish by the name what is the difference between this and the `validateAndSanitizeTLSSecret` helper.
  b. I do not understand the `Or` in the name. This function handles only one case
  c. Replace TlS by TLS in the name
3. I still think this function should not be part of the secretref package. It works with a map, but does not have anything to do with a Secret(-reference)

File: internal/reconciler/logpipeline/sync.go [hunk 4]
```diff
@@ -197,6 +199,12 @@ func (s *syncer) syncTLSConfigSecret(ctx context.Context, logPipelines []telemet
 		if err = controllerutil.SetOwnerReference(&logPipelines[i], &newSecret, s.Scheme()); err != nil {
 			return fmt.Errorf("unable to set owner reference for tls config secret: %w", err)
 		}
+
+		if tls.Cert.IsDefined() && tls.Key.IsDefined() {
```
Comment (chrkl) [inline internal/reconciler/logpipeline/sync.go:203]: Maybe let's change the control flow here similar to the Otel case.

File: internal/secretref/secret_ref.go (hunk 3 above)
Comment (rakesh-garimella) [reply internal/secretref/secret_ref.go]: I removed this function and simplified the code

File: internal/reconciler/logpipeline/sync.go (hunk 4 above)
Comment (rakesh-garimella) [reply internal/reconciler/logpipeline/sync.go]: done

**Review by chrkl — DISMISSED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**